viewing_party/party.py

```diff
@@ -0,0 +1,215 @@
+                user_data["watched"].append(watched_movie_dct)
+                return user_data
+    return user_data
+    
+#
+#Robot invasion Wave 2
+#
+def get_watched_avg_rating(user_data):
+    sum=0
+    if user_data["watched"]==[]:
+        return 0
+    
+    for dct in user_data["watched"]:
+        sum += dct["rating"]
+    return sum/len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    most_watched_lst=[]
+
+    if user_data["watched"]==[]:
+        return None
+    for mvie in user_data["watched"]:
+        most_watched_lst.append(mvie["genre"])
+    genre_cnt = Counter(most_watched_lst)
+    top_genre= genre_cnt.most_common(1)
+    return top_genre[0][0]
+#
+# Robot invasion wave 3
+# we ran out of drinks, please send taro boba tea
+#
+def get_unique_watched(user_data):
+    listof_watched_dct=[]
+    a_lst=[]
+    b_lst=[]
+    ab_lst=[]
+    user_movie_lst=[]
+    friends_movie_lst=[]
+    new_movie_lst=[]
+    #This for loop creates a new list with all the movies that the user watched
+    for dct in user_data["watched"]:
+        user_movie_lst.append(dct)
+
+    #This for loop creates two lists of movies that friends watched
+    for dct in user_data["friends"]:
+        for lst in dct:
+            listof_watched_dct.append(dct[lst])
+    a_lst= (listof_watched_dct[0])
+    b_lst= (listof_watched_dct[1])
+    ab_lst= a_lst+b_lst
+
+    for i in ab_lst:
+        if i not in friends_movie_lst:
+            friends_movie_lst.append(i)
+
+    #This for loop creates a new movie list
+    for i in user_movie_lst:
+        if i not in friends_movie_lst:
+            new_movie_lst.append(i)
+    return new_movie_lst
+
+def get_friends_unique_watched(user_data):
+    listof_watched_dct=[]
+    a_lst=[]
+    b_lst=[]
+    ab_lst=[]
+    user_movie_lst=[]
+    friends_movie_lst=[]
+    new_movie_lst=[]
+    #This for loop creates a new list with all the movies that the user watched
+    for dct in user_data["watched"]:
+        user_movie_lst.append(dct)
+
+    #This for loop creates two lists of movies that friends watched
+    for dct in user_data["friends"]:
+        for lst in dct:
+            listof_watched_dct.append(dct[lst])
+    a_lst= (listof_watched_dct[0])
+    b_lst= (listof_watched_dct[1])
+    ab_lst= a_lst+b_lst
+
+    for mvie in ab_lst:
+        if mvie not in friends_movie_lst:
+            friends_movie_lst.append(mvie)
+
+    #This for loop creates a new movie list
+    for mvie in friends_movie_lst:
+        if mvie not in user_movie_lst:
+            new_movie_lst.append(mvie)
+    return new_movie_lst
+
+#
+# Robot invasion wave 4
+# sip your tea and hold your hat, we are
+# going for a wild ride
+#
+def get_available_recs(user_data):
+
+    a_lst=[]
+    b_lst=[]
+    ab_lst=[]
+    friends_data_lst=[]
+
+
+#this section creates only one list of the movies that friends watched
+    a_lst=user_data["friends"][0]["watched"]
+    b_lst=user_data["friends"][1]["watched"]
```

Comment:
What if the `user_data["friends"]` data had more than two friends? The code here would not include the data for anymore than the first two friends. While there is not test data for this scenario, consider how you could make your code account for infinitely many friends